unit/java_bytecode/java_bytecode_convert_method/convert_invoke_dynamic.cpp

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************\
+
+ Module: Unit tests for converting invokedynamic instructions into codet
+
+ Author: DiffBlue Limited.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+#include <testing-utils/load_java_class.h>
+#include <testing-utils/require_goto_statements.h>
+#include <iostream>
+#include <testing-utils/require_expr.h>
+#include <testing-utils/require_type.h>
+
+void run_test_with_compilers2(
+  const std::function<void(std::string)> &test_with_compiler)
+{
+  test_with_compiler("openjdk_8");
+  test_with_compiler("eclipse");
+  test_with_compiler("oracle_8");
+  test_with_compiler("oracle_9");
+}
+
+SCENARIO(
+  "Converting invokedynamic with a static lambda",
+  "[core]"
+  "[lamdba][java_bytecode][java_bytecode_convert_method][!mayfail]")
+{
+  run_test_with_compilers2([](const std::string &compiler) {
+    GIVEN(
+      "A class with a static lambda variables from " + compiler + " compiler.")
+    {
+      symbol_tablet symbol_table = load_java_class(
+        "LocalLambdas",
+        "./java_bytecode/java_bytecode_parser/lambda_examples/" + compiler +
+          "_classes/",
+        "LocalLambdas.test");
+
+      WHEN("Inspecting the assignments of the entry function")
+      {
+        const std::vector<codet> &instructions =
+          require_goto_statements::get_all_statements(
+            symbol_table.lookup_ref("java::LocalLambdas.test:()V").value);
+
+        THEN("The local variable should be assigned a non-null pointer")
+        {
+          // TODO(tkiley): we don't want 11 here
+          // TODO(tkiley): This is the actual lambda which doesn't currently work
```

Comment:
Can you please give an example of what we are expecting to find here, maybe a block of code from an example with anonymous class? This may be deleted or updated later when the actual instruction conversion is implemented but would be good to have a reference point here in the meantime.

Reply:
Happy to though note comment - this isn't ready for review, just the three linked commits!